tests/output.rs

```diff
@@ -675,3 +675,42 @@ fn test_macro_contents() {
         "#,
     );
 }
+
+#[test]
+fn test_sorting_maybe_nesting_related() {
```

Comment:
So I closed #43 because my understanding was that by switching to the "captures-oriented" query API + maybe some of the custom `grep` stuff we would be guaranteed to not have this ordering issue, but this branch was based on that branch so the fact that the test case from that branch passes (without the explicit `.collect()`'ing + `.sort()`'ing from that branch) is a good sign